Client/Frontend/Sync/BraveCore/Bookmarks/BookmarkManager.swift

```diff
@@ -0,0 +1,303 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Data
+import BraveCore
+import BraveShared
+import CoreData
+import Shared
+
+private let log = Logger.browserLogger
+
+class BookmarkManager: NSObject {
+
+    // MARK: Lifecycle
+    
+    init(bookmarksAPI: BraveBookmarksAPI?) {
+        self.bookmarksAPI = bookmarksAPI
+        super.init()
+        
+        BookmarkManager.rootNodeId = bookmarksAPI?.rootNode?.guid
```

Comment:
Why is this a statically set value? Just create a function in this manager class that takes a node returns a parent and remove the `parent` variable in `Bookmarksv2`